<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.7.32">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">


<title>Sepsis RNA array analysis</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for syntax highlighting */
html { -webkit-text-size-adjust: 100%; }
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
  }
pre.numberSource { margin-left: 3em;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
</style>


<script src="Sepsis_rna_array_analysis_GSE65682_files/libs/clipboard/clipboard.min.js"></script>
<script src="Sepsis_rna_array_analysis_GSE65682_files/libs/quarto-html/quarto.js" type="module"></script>
<script src="Sepsis_rna_array_analysis_GSE65682_files/libs/quarto-html/tabsets/tabsets.js" type="module"></script>
<script src="Sepsis_rna_array_analysis_GSE65682_files/libs/quarto-html/popper.min.js"></script>
<script src="Sepsis_rna_array_analysis_GSE65682_files/libs/quarto-html/tippy.umd.min.js"></script>
<script src="Sepsis_rna_array_analysis_GSE65682_files/libs/quarto-html/anchor.min.js"></script>
<link href="Sepsis_rna_array_analysis_GSE65682_files/libs/quarto-html/tippy.css" rel="stylesheet">
<link href="Sepsis_rna_array_analysis_GSE65682_files/libs/quarto-html/quarto-syntax-highlighting-37eea08aefeeee20ff55810ff984fec1.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="Sepsis_rna_array_analysis_GSE65682_files/libs/bootstrap/bootstrap.min.js"></script>
<link href="Sepsis_rna_array_analysis_GSE65682_files/libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="Sepsis_rna_array_analysis_GSE65682_files/libs/bootstrap/bootstrap-bb462d781dde1847d9e3ccf7736099dd.min.css" rel="stylesheet" append-hash="true" id="quarto-bootstrap" data-mode="light">


</head>

<body class="fullcontent quarto-light">

<div id="quarto-content" class="page-columns page-rows-contents page-layout-article">

<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<h1 class="title">Sepsis RNA array analysis</h1>
</div>



<div class="quarto-title-meta">

    
  
    
  </div>
  


</header>


<section id="rna-array-profiles-of-over-800-sepsis-patients-with-varying-survivor-status-data-from-gse65682-and-httpswww.sciencedirect.comsciencearticlepiis2213260017302941cesec20" class="level1">
<h1>RNA array profiles of over 800 sepsis patients with varying survivor status Data from GSE65682 and https://www.sciencedirect.com/science/article/pii/S2213260017302941#cesec20</h1>
<div class="cell">
<div class="sourceCode cell-code" id="cb1"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(GEOquery)</span>
<span id="cb1-2"><a href="#cb1-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(biomaRt)</span>
<span id="cb1-3"><a href="#cb1-3" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(limma)</span>
<span id="cb1-4"><a href="#cb1-4" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(dplyr)</span>
<span id="cb1-5"><a href="#cb1-5" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tidyr)</span>
<span id="cb1-6"><a href="#cb1-6" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggplot2)</span>
<span id="cb1-7"><a href="#cb1-7" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ranger)</span>
<span id="cb1-8"><a href="#cb1-8" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(pROC)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb2"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1" aria-hidden="true" tabindex="-1"></a><span class="fu">setwd</span>(<span class="st">"C:/Users/dan94/rips_project/"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<section id="loading-in-data" class="level2">
<h2 class="anchored" data-anchor-id="loading-in-data">LOADING IN DATA</h2>
<section id="load-in-the-expression-matrix" class="level3">
<h3 class="anchored" data-anchor-id="load-in-the-expression-matrix">load in the expression matrix</h3>
</section>
<section id="getgeo-pulls-directly-from-the-geo-website" class="level3">
<h3 class="anchored" data-anchor-id="getgeo-pulls-directly-from-the-geo-website">getGEO pulls directly from the GEO website</h3>
</section>
<section id="take-the-first-slot-of-the-data-containing-the-correct-platform" class="level3">
<h3 class="anchored" data-anchor-id="take-the-first-slot-of-the-data-containing-the-correct-platform">take the first slot of the data containing the correct platform</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb3"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1" aria-hidden="true" tabindex="-1"></a>geo_data <span class="ot">&lt;-</span> <span class="fu">getGEO</span>(<span class="st">"GSE65682"</span>, <span class="at">GSEMatrix =</span> T)</span>
<span id="cb3-2"><a href="#cb3-2" aria-hidden="true" tabindex="-1"></a>geo_data <span class="ot">&lt;-</span> geo_data[[<span class="dv">1</span>]]</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="extract-the-expression-data-and-accompanying-metadata" class="level3">
<h3 class="anchored" data-anchor-id="extract-the-expression-data-and-accompanying-metadata">extract the expression data and accompanying metadata</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb4"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1" aria-hidden="true" tabindex="-1"></a>exprs_df <span class="ot">&lt;-</span> <span class="fu">exprs</span>(geo_data)</span>
<span id="cb4-2"><a href="#cb4-2" aria-hidden="true" tabindex="-1"></a>exprs_df <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(exprs_df)</span>
<span id="cb4-3"><a href="#cb4-3" aria-hidden="true" tabindex="-1"></a>exprs_df<span class="sc">$</span>probe_id <span class="ot">&lt;-</span> <span class="fu">rownames</span>(exprs_df)</span>
<span id="cb4-4"><a href="#cb4-4" aria-hidden="true" tabindex="-1"></a>sample_data <span class="ot">&lt;-</span> <span class="fu">pData</span>(geo_data)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
</section>
<section id="getting-gene-names-for-probes" class="level2">
<h2 class="anchored" data-anchor-id="getting-gene-names-for-probes">GETTING GENE NAMES FOR PROBES</h2>
<section id="we-access-the-gpl-number-from-same-page-the-geo-object-was-obtained" class="level3">
<h3 class="anchored" data-anchor-id="we-access-the-gpl-number-from-same-page-the-geo-object-was-obtained">we access the gpl number from same page the geo object was obtained</h3>
</section>
<section id="find-the-appropraite-columns-id-and-gene-symbol-and-make-a-df" class="level3">
<h3 class="anchored" data-anchor-id="find-the-appropraite-columns-id-and-gene-symbol-and-make-a-df">find the appropraite columns (ID and Gene symbol) and make a df</h3>
<section id="add-this-df-to-the-exprs-df" class="level4">
<h4 class="anchored" data-anchor-id="add-this-df-to-the-exprs-df">add this df to the exprs df</h4>
</section>
</section>
<section id="changing-genes-with-two-names-and-to-just-the-first-name" class="level3">
<h3 class="anchored" data-anchor-id="changing-genes-with-two-names-and-to-just-the-first-name">changing gene’s with two names and “///” to just the first name</h3>
</section>
<section id="finally-to-handle-duplicate-genes-presumably-because-an-array-can-include-multiple-probes-to-the-same-gene-average-them" class="level3">
<h3 class="anchored" data-anchor-id="finally-to-handle-duplicate-genes-presumably-because-an-array-can-include-multiple-probes-to-the-same-gene-average-them">finally, to handle duplicate genes (presumably because an array can include multiple probes to the same gene) average them</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb5"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1" aria-hidden="true" tabindex="-1"></a>gpl <span class="ot">&lt;-</span> <span class="fu">getGEO</span>(<span class="st">"GPL13667"</span>)</span>
<span id="cb5-2"><a href="#cb5-2" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(<span class="fu">Table</span>(gpl))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>             ID             GeneChip Array Species Scientific Name
1   11715100_at Human Genome HG-U219 Array            Homo sapiens
2 11715101_s_at Human Genome HG-U219 Array            Homo sapiens
3 11715102_x_at Human Genome HG-U219 Array            Homo sapiens
4 11715103_x_at Human Genome HG-U219 Array            Homo sapiens
5 11715104_s_at Human Genome HG-U219 Array            Homo sapiens
6   11715105_at Human Genome HG-U219 Array            Homo sapiens
  Annotation Date      Sequence Type                 Sequence Source
1       20-Aug-10 Consensus sequence Affymetrix Proprietary Database
2       20-Aug-10 Consensus sequence Affymetrix Proprietary Database
3       20-Aug-10 Consensus sequence Affymetrix Proprietary Database
4       20-Aug-10 Consensus sequence Affymetrix Proprietary Database
5       20-Aug-10 Consensus sequence Affymetrix Proprietary Database
6       20-Aug-10 Consensus sequence Affymetrix Proprietary Database
  Transcript ID(Array Design)
1                   g21264570
2                   g21264570
3                   g21264570
4                   g22748780
5                   g30039713
6                   g45580703
                                                                                Target Description
1 g21264570 /TID=g21264570 /CNT=1 /FEA=FLmRNA /TIER=FL /STK=0 /DEF=g21264570 /REP_ORG=Homo sapiens
2 g21264570 /TID=g21264570 /CNT=1 /FEA=FLmRNA /TIER=FL /STK=0 /DEF=g21264570 /REP_ORG=Homo sapiens
3 g21264570 /TID=g21264570 /CNT=1 /FEA=FLmRNA /TIER=FL /STK=0 /DEF=g21264570 /REP_ORG=Homo sapiens
4 g22748780 /TID=g22748780 /CNT=1 /FEA=FLmRNA /TIER=FL /STK=0 /DEF=g22748780 /REP_ORG=Homo sapiens
5 g30039713 /TID=g30039713 /CNT=1 /FEA=FLmRNA /TIER=FL /STK=0 /DEF=g30039713 /REP_ORG=Homo sapiens
6 g45580703 /TID=g45580703 /CNT=1 /FEA=FLmRNA /TIER=FL /STK=0 /DEF=g45580703 /REP_ORG=Homo sapiens
  Representative Public ID Archival UniGene Cluster UniGene ID
1                g21264570                      ---  Hs.247813
2                g21264570                      ---  Hs.247813
3                g21264570                      ---  Hs.247813
4                g22748780                      ---  Hs.465643
5                g30039713                      ---  Hs.352515
6                g45580703                      ---  Hs.439154
                                      Genome Version
1 February 2009 (Genome Reference Consortium GRCh37)
2 February 2009 (Genome Reference Consortium GRCh37)
3 February 2009 (Genome Reference Consortium GRCh37)
4 February 2009 (Genome Reference Consortium GRCh37)
5 February 2009 (Genome Reference Consortium GRCh37)
6 February 2009 (Genome Reference Consortium GRCh37)
                                     Alignments
1  chr6:26271145-26271612 (-) // 100.0 // p22.2
2  chr6:26271145-26271612 (-) // 100.0 // p22.2
3  chr6:26271145-26271612 (-) // 100.0 // p22.2
4   chr19:4639529-5145579 (+) // 48.53 // p13.3
5 chr17:72920369-72929640 (+) // 100.0 // q25.1
6   chr17:35732984-35749660 (+) // 98.15 // q12
                                             Gene Title Gene Symbol
1                                histone cluster 1, H3g    HIST1H3G
2                                histone cluster 1, H3g    HIST1H3G
3                                histone cluster 1, H3g    HIST1H3G
4 tumor necrosis factor, alpha-induced protein 8-like 1   TNFAIP8L1
5                                           otopetrin 2       OTOP2
6                   chromosome 17 open reading frame 78    C17orf78
  Chromosomal Location                GB_LIST SPOT_ID Unigene Cluster Type
1            chr6p21.3              NM_003534                  full length
2            chr6p21.3              NM_003534                  full length
3            chr6p21.3              NM_003534                  full length
4           chr19p13.3 NM_001167942,NM_152362                  full length
5           chr17q25.1              NM_178160                  full length
6             chr17q12              NM_173625                  full length
          Ensembl Entrez Gene SwissProt  EC   OMIM          RefSeq Protein ID
1             ---        8355    P68431 --- 602815                  NP_003525
2 ENSG00000178458        8355    P68431 --- 602815                  NP_003525
3             ---        8355    P68431 --- 602815                  NP_003525
4 ENSG00000185361      126282    Q8WVP5 ---    --- NP_001161414 /// NP_689575
5 ENSG00000183034       92736    Q7RTS6 --- 607827                  NP_835454
6 ENSG00000167230      284099    Q8N4C9 ---    ---                  NP_775896
        RefSeq Transcript ID FlyBase AGI WormBase MGI Name RGD Name
1                  NM_003534     --- ---      ---      ---      ---
2                  NM_003534     --- ---      ---      ---      ---
3                  NM_003534     --- ---      ---      ---      ---
4 NM_001167942 /// NM_152362     --- ---      ---      ---      ---
5                  NM_178160     --- ---      ---      ---      ---
6                  NM_173625     --- ---      ---      ---      ---
  SGD accession number
1                  ---
2                  ---
3                  ---
4                  ---
5                  ---
6                  ---
                                       Gene Ontology Biological Process
1 0006334 // nucleosome assembly // inferred from electronic annotation
2 0006334 // nucleosome assembly // inferred from electronic annotation
3 0006334 // nucleosome assembly // inferred from electronic annotation
4                                                                   ---
5                                                                   ---
6                                                                   ---
                                                                                                                                                             Gene Ontology Cellular Component
1 0000786 // nucleosome // inferred from electronic annotation /// 0005634 // nucleus // inferred from electronic annotation /// 0005694 // chromosome // inferred from electronic annotation
2 0000786 // nucleosome // inferred from electronic annotation /// 0005634 // nucleus // inferred from electronic annotation /// 0005694 // chromosome // inferred from electronic annotation
3 0000786 // nucleosome // inferred from electronic annotation /// 0005634 // nucleus // inferred from electronic annotation /// 0005694 // chromosome // inferred from electronic annotation
4                                                                                                                                                                                         ---
5                                                       0016020 // membrane // inferred from electronic annotation /// 0016021 // integral to membrane // inferred from electronic annotation
6                                                       0016020 // membrane // inferred from electronic annotation /// 0016021 // integral to membrane // inferred from electronic annotation
                                                                                                    Gene Ontology Molecular Function
1 0003677 // DNA binding // inferred from electronic annotation /// 0005515 // protein binding // inferred from physical interaction
2 0003677 // DNA binding // inferred from electronic annotation /// 0005515 // protein binding // inferred from physical interaction
3 0003677 // DNA binding // inferred from electronic annotation /// 0005515 // protein binding // inferred from physical interaction
4                                                                                                                                ---
5                                                                                                                                ---
6                                                                                                                                ---
  Pathway
1     ---
2     ---
3     ---
4     ---
5     ---
6     ---
                                                                                                                  InterPro
1                                                                                                                      ---
2                                                                                                                      ---
3                                                                                                                      ---
4                                                                                                                      ---
5 IPR004878 // Protein of unknown function DUF270 // 1.0E-6 /// IPR004878 // Protein of unknown function DUF270 // 1.0E-13
6                                                                                                                      ---
                                                                                               Trans Membrane
1                                                                                                         ---
2                                                                                                         ---
3                                                                                                         ---
4                                                                                                         ---
5 NP_835454.1 // span:30-52,62-81,101-120,135-157,240-262,288-310,327-349,369-391,496-515,525-547 // numtm:10
6                            AAH34672.1 // span:184-206 // numtm:1 /// NP_775896.3 // span:184-206 // numtm:1
  QTL
1 ---
2 ---
3 ---
4 ---
5 ---
6 ---
                                                                                                                                   Annotation Description
1 This probe set was annotated using the Matching Probes based pipeline to a Entrez Gene identifier using 1 transcripts. // false // Matching Probes // A
2 This probe set was annotated using the Matching Probes based pipeline to a Entrez Gene identifier using 2 transcripts. // false // Matching Probes // A
3 This probe set was annotated using the Matching Probes based pipeline to a Entrez Gene identifier using 1 transcripts. // false // Matching Probes // A
4 This probe set was annotated using the Matching Probes based pipeline to a Entrez Gene identifier using 5 transcripts. // false // Matching Probes // A
5 This probe set was annotated using the Matching Probes based pipeline to a Entrez Gene identifier using 3 transcripts. // false // Matching Probes // A
6 This probe set was annotated using the Matching Probes based pipeline to a Entrez Gene identifier using 5 transcripts. // false // Matching Probes // A
                                                    Annotation Transcript Cluster
1                                                                   NM_003534(11)
2                                                      BC079835(11),NM_003534(11)
3                                                                   NM_003534(11)
4     BC017672(11),BC044250(9),ENST00000327473(11),NM_001167942(11),NM_152362(11)
5                           ENST00000331427(11),ENST00000426069(11),NM_178160(11)
6 AK096966(10),BC034672(11),ENST00000300618(11),ENST00000321564(11),NM_173625(11)
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                               Transcript Assignments
1                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                           NM_003534 // Homo sapiens histone cluster 1, H3g (HIST1H3G), mRNA. // refseq // 11 // ---
2                                                                                                                                                                                                                                                                                              BC079835 // Homo sapiens histone cluster 1, H3g, mRNA (cDNA clone IMAGE:5935692). // gb_htc // 11 // --- /// ENST00000321285 // cdna:known chromosome:GRCh37:6:26271202:26271612:-1 gene:ENSG00000178458 // ensembl // 11 // --- /// GENSCAN00000044911 // cdna:Genscan chromosome:GRCh37:6:26271202:26271612:-1 // ensembl // 11 // --- /// NM_003534 // Homo sapiens histone cluster 1, H3g (HIST1H3G), mRNA. // refseq // 11 // ---
3                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                           NM_003534 // Homo sapiens histone cluster 1, H3g (HIST1H3G), mRNA. // refseq // 11 // ---
4 BC017672 // Homo sapiens tumor necrosis factor, alpha-induced protein 8-like 1, mRNA (cDNA clone MGC:17791 IMAGE:3885999), complete cds. // gb // 11 // --- /// BC044250 // Homo sapiens tumor necrosis factor, alpha-induced protein 8-like 1, mRNA (cDNA clone IMAGE:5784807). // gb // 9 // --- /// ENST00000327473 // cdna:known chromosome:GRCh37:19:4639530:4653952:1 gene:ENSG00000185361 // ensembl // 11 // --- /// NM_001167942 // Homo sapiens tumor necrosis factor, alpha-induced protein 8-like 1 (TNFAIP8L1), transcript variant 1, mRNA. // refseq // 11 // --- /// NM_152362 // Homo sapiens tumor necrosis factor, alpha-induced protein 8-like 1 (TNFAIP8L1), transcript variant 2, mRNA. // refseq // 11 // ---
5                                                                                                                                                                                                                                                                                                                                                                                                         ENST00000331427 // cdna:known chromosome:GRCh37:17:72920370:72929640:1 gene:ENSG00000183034 // ensembl // 11 // --- /// ENST00000426069 // cdna:known chromosome:GRCh37:17:72920370:72929640:1 gene:ENSG00000183034 // ensembl // 11 // --- /// NM_178160 // Homo sapiens otopetrin 2 (OTOP2), mRNA. // refseq // 11 // ---
6                                                                                                                                         AK096966 // Homo sapiens cDNA FLJ39647 fis, clone SMINT2004781. // gb // 10 // --- /// BC034672 // Homo sapiens chromosome 17 open reading frame 78, mRNA (cDNA clone MGC:34759 IMAGE:5184947), complete cds. // gb // 11 // --- /// ENST00000300618 // cdna:known chromosome:GRCh37:17:35732985:35749660:1 gene:ENSG00000167230 // ensembl // 11 // --- /// ENST00000321564 // cdna:known chromosome:GRCh37:17:35732985:35749660:1 gene:ENSG00000167230 // ensembl // 11 // --- /// NM_173625 // Homo sapiens chromosome 17 open reading frame 78 (C17orf78), mRNA. // refseq // 11 // ---
                                                                                                                                                                           Annotation Notes
1                                                                                                                                      BC079835 // gb_htc // 6 // Cross Hyb Matching Probes
2                                                                                                                                                                                       ---
3 GENSCAN00000044911 // ensembl // 4 // Cross Hyb Matching Probes /// ENST00000321285 // ensembl // 4 // Cross Hyb Matching Probes /// BC079835 // gb_htc // 7 // Cross Hyb Matching Probes
4                                                                                                                                                                                       ---
5                                                                                                                           GENSCAN00000031612 // ensembl // 8 // Cross Hyb Matching Probes
6                                                                                                                                                                                       ---</code></pre>
</div>
<div class="sourceCode cell-code" id="cb7"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb7-1"><a href="#cb7-1" aria-hidden="true" tabindex="-1"></a>probes <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">probe_id =</span> <span class="fu">Table</span>(gpl)[[<span class="st">"ID"</span>]], <span class="at">gene_id =</span> <span class="fu">Table</span>(gpl)[[<span class="st">"Gene Symbol"</span>]])</span>
<span id="cb7-2"><a href="#cb7-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-3"><a href="#cb7-3" aria-hidden="true" tabindex="-1"></a>exprs_df <span class="ot">&lt;-</span> <span class="fu">merge</span>(exprs_df, probes, <span class="at">by =</span> <span class="st">"probe_id"</span>)</span>
<span id="cb7-4"><a href="#cb7-4" aria-hidden="true" tabindex="-1"></a>exprs_df <span class="ot">&lt;-</span> exprs_df[, <span class="sc">-</span><span class="dv">1</span>]</span>
<span id="cb7-5"><a href="#cb7-5" aria-hidden="true" tabindex="-1"></a>exprs_df<span class="sc">$</span>gene_id <span class="ot">&lt;-</span> <span class="fu">sub</span>(<span class="st">" ///.*"</span>, <span class="st">""</span>, exprs_df<span class="sc">$</span>gene_id)</span>
<span id="cb7-6"><a href="#cb7-6" aria-hidden="true" tabindex="-1"></a>exprs_df <span class="ot">&lt;-</span> exprs_df <span class="sc">%&gt;%</span></span>
<span id="cb7-7"><a href="#cb7-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="sc">!</span><span class="fu">is.na</span>(gene_id) <span class="sc">&amp;</span> gene_id <span class="sc">!=</span> <span class="st">""</span>) <span class="sc">%&gt;%</span></span>
<span id="cb7-8"><a href="#cb7-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(gene_id) <span class="sc">%&gt;%</span></span>
<span id="cb7-9"><a href="#cb7-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarise</span>(<span class="fu">across</span>(<span class="fu">where</span>(is.numeric), <span class="sc">~</span> <span class="fu">mean</span>(.x, <span class="at">na.rm =</span> <span class="cn">TRUE</span>)), <span class="at">.groups =</span> <span class="st">"drop"</span>)</span>
<span id="cb7-10"><a href="#cb7-10" aria-hidden="true" tabindex="-1"></a><span class="co">#rownames(exprs_df) &lt;- exprs_df$gene_id</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
</section>
<section id="setting-up-data" class="level2">
<h2 class="anchored" data-anchor-id="setting-up-data">SETTING UP DATA</h2>
<section id="transpose-the-expression-df-to-have-samples-as-a-row" class="level3">
<h3 class="anchored" data-anchor-id="transpose-the-expression-df-to-have-samples-as-a-row">transpose the expression df to have samples as a row</h3>
</section>
<section id="this-can-then-be-merged-with-the-metadata" class="level3">
<h3 class="anchored" data-anchor-id="this-can-then-be-merged-with-the-metadata">this can then be merged with the metadata</h3>
</section>
<section id="this-results-in-a-single-df-with-each-patient-in-a-row-with-their-expression-data-and-clinical-info-in-the-columns" class="level3">
<h3 class="anchored" data-anchor-id="this-results-in-a-single-df-with-each-patient-in-a-row-with-their-expression-data-and-clinical-info-in-the-columns">this results in a single df with each patient in a row, with their expression data and clinical info in the columns</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb8"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb8-1"><a href="#cb8-1" aria-hidden="true" tabindex="-1"></a>full_df <span class="ot">&lt;-</span> <span class="fu">pivot_longer</span>(exprs_df, <span class="at">names_to =</span> <span class="st">"sample_id"</span>, <span class="at">values_to =</span> <span class="st">"expression"</span>, <span class="at">cols =</span> <span class="dv">2</span><span class="sc">:</span><span class="dv">803</span>)</span>
<span id="cb8-2"><a href="#cb8-2" aria-hidden="true" tabindex="-1"></a>full_df <span class="ot">&lt;-</span> <span class="fu">pivot_wider</span>(full_df, <span class="at">names_from =</span> gene_id, <span class="at">values_from =</span> expression)</span>
<span id="cb8-3"><a href="#cb8-3" aria-hidden="true" tabindex="-1"></a>full_df <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(full_df)</span>
<span id="cb8-4"><a href="#cb8-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-5"><a href="#cb8-5" aria-hidden="true" tabindex="-1"></a>sample_data<span class="sc">$</span>sample_id <span class="ot">&lt;-</span> <span class="fu">rownames</span>(sample_data)</span>
<span id="cb8-6"><a href="#cb8-6" aria-hidden="true" tabindex="-1"></a>full_df <span class="ot">&lt;-</span> <span class="fu">merge</span>(full_df, sample_data, <span class="at">by =</span> <span class="st">"sample_id"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
</section>
<section id="plotting-gene-expression" class="level2">
<h2 class="anchored" data-anchor-id="plotting-gene-expression">PLOTTING GENE EXPRESSION</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb9"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb9-1"><a href="#cb9-1" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(full_df, <span class="fu">aes</span>(<span class="at">x =</span> <span class="st">`</span><span class="at">mortality_event_28days:ch1</span><span class="st">`</span>, <span class="at">y =</span> ETS2, <span class="at">fill =</span> <span class="st">`</span><span class="at">endotype_cohort:ch1</span><span class="st">`</span>))<span class="sc">+</span><span class="fu">geom_boxplot</span>(<span class="at">outlier.shape =</span> <span class="cn">NA</span>)<span class="sc">+</span><span class="fu">theme_bw</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="Sepsis_rna_array_analysis_GSE65682_files/figure-html/unnamed-chunk-7-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
</section>
<section id="save-full-dataframe" class="level2">
<h2 class="anchored" data-anchor-id="save-full-dataframe">SAVE FULL DATAFRAME</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb10"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb10-1"><a href="#cb10-1" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(full_df, <span class="at">file =</span> <span class="st">"sepsis_rna_array.rds"</span>, <span class="at">compress =</span> F)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="random-forest-to-distinguish-samples-by-mortality" class="level2">
<h2 class="anchored" data-anchor-id="random-forest-to-distinguish-samples-by-mortality">RANDOM FOREST TO DISTINGUISH SAMPLES BY MORTALITY</h2>
<section id="select-only-gene-columns-plus-grouping-variable-of-interest" class="level3">
<h3 class="anchored" data-anchor-id="select-only-gene-columns-plus-grouping-variable-of-interest">select only gene columns plus grouping variable of interest</h3>
</section>
<section id="renaming-mortality-column-for-ease-and-removing-old-name" class="level3">
<h3 class="anchored" data-anchor-id="renaming-mortality-column-for-ease-and-removing-old-name">renaming mortality column for ease and removing old name</h3>
</section>
<section id="make-all-gene-columns-numeric" class="level3">
<h3 class="anchored" data-anchor-id="make-all-gene-columns-numeric">make all gene columns numeric</h3>
</section>
<section id="then-make-grouping-variable-a-factor" class="level3">
<h3 class="anchored" data-anchor-id="then-make-grouping-variable-a-factor">then make grouping variable a factor</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb11"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb11-1"><a href="#cb11-1" aria-hidden="true" tabindex="-1"></a>rf_data <span class="ot">&lt;-</span> full_df[, <span class="fu">c</span>(<span class="dv">3</span><span class="sc">:</span><span class="dv">11520</span>,<span class="dv">11564</span>,<span class="dv">11568</span>,<span class="dv">11571</span>,<span class="dv">11572</span>)]</span>
<span id="cb11-2"><a href="#cb11-2" aria-hidden="true" tabindex="-1"></a>rf_data<span class="sc">$</span>mortality_28d <span class="ot">&lt;-</span> rf_data<span class="sc">$</span><span class="st">`</span><span class="at">mortality_event_28days:ch1</span><span class="st">`</span></span>
<span id="cb11-3"><a href="#cb11-3" aria-hidden="true" tabindex="-1"></a>rf_data <span class="ot">&lt;-</span> rf_data[, <span class="sc">-</span><span class="dv">11521</span>]</span>
<span id="cb11-4"><a href="#cb11-4" aria-hidden="true" tabindex="-1"></a>rf_data[, <span class="dv">1</span><span class="sc">:</span><span class="dv">11518</span>] <span class="ot">&lt;-</span> <span class="fu">lapply</span>(rf_data[, <span class="dv">1</span><span class="sc">:</span><span class="dv">11518</span>], as.numeric)</span>
<span id="cb11-5"><a href="#cb11-5" aria-hidden="true" tabindex="-1"></a>rf_data[, <span class="dv">11519</span><span class="sc">:</span><span class="dv">11522</span>] <span class="ot">&lt;-</span> <span class="fu">lapply</span>(rf_data[, <span class="dv">11519</span><span class="sc">:</span><span class="dv">11522</span>], as.factor)</span>
<span id="cb11-6"><a href="#cb11-6" aria-hidden="true" tabindex="-1"></a>rf_data <span class="ot">&lt;-</span> <span class="fu">filter</span>(rf_data, mortality_28d <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">"0"</span>,<span class="st">"1"</span>))</span>
<span id="cb11-7"><a href="#cb11-7" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(rf_data) <span class="ot">&lt;-</span> <span class="fu">sub</span>(<span class="st">"-"</span>, <span class="st">"_"</span>, <span class="fu">colnames</span>(rf_data))</span>
<span id="cb11-8"><a href="#cb11-8" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(rf_data) <span class="ot">&lt;-</span> <span class="fu">sub</span>(<span class="st">"@"</span>, <span class="st">""</span>, <span class="fu">colnames</span>(rf_data))</span>
<span id="cb11-9"><a href="#cb11-9" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(rf_data) <span class="ot">&lt;-</span> <span class="fu">sub</span>(<span class="st">" "</span>, <span class="st">""</span>, <span class="fu">colnames</span>(rf_data))</span>
<span id="cb11-10"><a href="#cb11-10" aria-hidden="true" tabindex="-1"></a>rf_data<span class="sc">$</span>mortality_28d <span class="ot">&lt;-</span> <span class="fu">droplevels</span>(rf_data<span class="sc">$</span>mortality_28d)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
</section>
<section id="run-random-forest" class="level2">
<h2 class="anchored" data-anchor-id="run-random-forest">run random forest</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb12"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb12-1"><a href="#cb12-1" aria-hidden="true" tabindex="-1"></a>y <span class="ot">&lt;-</span> rf_data<span class="sc">$</span>mortality_28d</span>
<span id="cb12-2"><a href="#cb12-2" aria-hidden="true" tabindex="-1"></a>x <span class="ot">&lt;-</span> rf_data[, <span class="sc">-</span><span class="dv">11522</span>]</span>
<span id="cb12-3"><a href="#cb12-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb12-4"><a href="#cb12-4" aria-hidden="true" tabindex="-1"></a><span class="fu">set.seed</span>(<span class="dv">61</span>)</span>
<span id="cb12-5"><a href="#cb12-5" aria-hidden="true" tabindex="-1"></a>rf_model <span class="ot">&lt;-</span> rf_model <span class="ot">&lt;-</span> <span class="fu">ranger</span>(</span>
<span id="cb12-6"><a href="#cb12-6" aria-hidden="true" tabindex="-1"></a>  <span class="at">x =</span> x, <span class="at">y =</span> y, </span>
<span id="cb12-7"><a href="#cb12-7" aria-hidden="true" tabindex="-1"></a>  <span class="at">num.trees =</span> <span class="dv">5000</span>,</span>
<span id="cb12-8"><a href="#cb12-8" aria-hidden="true" tabindex="-1"></a>  <span class="at">importance =</span> <span class="st">"permutation"</span>,</span>
<span id="cb12-9"><a href="#cb12-9" aria-hidden="true" tabindex="-1"></a>  <span class="at">probability =</span> <span class="cn">TRUE</span>,                    <span class="co"># needed for ROC/AUC</span></span>
<span id="cb12-10"><a href="#cb12-10" aria-hidden="true" tabindex="-1"></a>  <span class="at">class.weights =</span> <span class="fu">c</span>(<span class="st">"0"</span> <span class="ot">=</span> <span class="dv">1</span>, <span class="st">"1"</span> <span class="ot">=</span> <span class="dv">4</span>)  <span class="co"># weight minority class</span></span>
<span id="cb12-11"><a href="#cb12-11" aria-hidden="true" tabindex="-1"></a>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="model-diagnostics" class="level2">
<h2 class="anchored" data-anchor-id="model-diagnostics">model diagnostics</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb13"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb13-1"><a href="#cb13-1" aria-hidden="true" tabindex="-1"></a>rf_model</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>Ranger result

Call:
 ranger(x = x, y = y, num.trees = 5000, importance = "permutation",      probability = TRUE, class.weights = c(`0` = 1, `1` = 4)) 

Type:                             Probability estimation 
Number of trees:                  5000 
Sample size:                      479 
Number of independent variables:  11521 
Mtry:                             107 
Target node size:                 10 
Variable importance mode:         permutation 
Splitrule:                        gini 
OOB prediction error (Brier s.):  0.1762718 </code></pre>
</div>
<div class="sourceCode cell-code" id="cb15"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb15-1"><a href="#cb15-1" aria-hidden="true" tabindex="-1"></a>rf_pred <span class="ot">&lt;-</span> rf_model<span class="sc">$</span>predictions[, <span class="st">"1"</span>]</span>
<span id="cb15-2"><a href="#cb15-2" aria-hidden="true" tabindex="-1"></a>roc_obj <span class="ot">&lt;-</span> <span class="fu">roc</span>(rf_data<span class="sc">$</span>mortality_28d, rf_pred)</span>
<span id="cb15-3"><a href="#cb15-3" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(roc_obj, <span class="at">col =</span> <span class="st">"blue"</span>, <span class="at">lwd =</span> <span class="dv">2</span>, <span class="at">main =</span> <span class="st">"ROC Curve - Ranger RF"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="Sepsis_rna_array_analysis_GSE65682_files/figure-html/unnamed-chunk-11-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="sourceCode cell-code" id="cb16"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb16-1"><a href="#cb16-1" aria-hidden="true" tabindex="-1"></a>auc_val <span class="ot">&lt;-</span> <span class="fu">auc</span>(roc_obj)</span>
<span id="cb16-2"><a href="#cb16-2" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(auc_val)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>Area under the curve: 0.6329</code></pre>
</div>
</div>
</section>
<section id="view-important-features-interactions" class="level2">
<h2 class="anchored" data-anchor-id="view-important-features-interactions">View important features (interactions)</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb18"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb18-1"><a href="#cb18-1" aria-hidden="true" tabindex="-1"></a>vi_df <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(</span>
<span id="cb18-2"><a href="#cb18-2" aria-hidden="true" tabindex="-1"></a>  <span class="at">Gene =</span> <span class="fu">names</span>(rf_model<span class="sc">$</span>variable.importance),</span>
<span id="cb18-3"><a href="#cb18-3" aria-hidden="true" tabindex="-1"></a>  <span class="at">Importance =</span> rf_model<span class="sc">$</span>variable.importance</span>
<span id="cb18-4"><a href="#cb18-4" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb18-5"><a href="#cb18-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-6"><a href="#cb18-6" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(vi_df[<span class="fu">order</span>(vi_df<span class="sc">$</span>Importance, <span class="at">decreasing =</span> <span class="cn">TRUE</span>)[<span class="dv">1</span><span class="sc">:</span><span class="dv">20</span>], ], </span>
<span id="cb18-7"><a href="#cb18-7" aria-hidden="true" tabindex="-1"></a>       <span class="fu">aes</span>(<span class="at">x =</span> <span class="fu">reorder</span>(Gene, Importance), <span class="at">y =</span> Importance, <span class="at">fill =</span> Importance)) <span class="sc">+</span></span>
<span id="cb18-8"><a href="#cb18-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">shape =</span> <span class="dv">21</span>, <span class="at">size =</span> <span class="dv">6</span>)<span class="sc">+</span></span>
<span id="cb18-9"><a href="#cb18-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">coord_flip</span>() <span class="sc">+</span></span>
<span id="cb18-10"><a href="#cb18-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">title =</span> <span class="st">"Top 20 Important Genes"</span>, <span class="at">x =</span> <span class="st">""</span>, <span class="at">y =</span> <span class="st">"Importance"</span>)<span class="sc">+</span></span>
<span id="cb18-11"><a href="#cb18-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_bw</span>()<span class="sc">+</span><span class="fu">theme</span>(<span class="at">axis.text =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">14</span>, <span class="at">colour =</span> <span class="st">"black"</span>))<span class="sc">+</span></span>
<span id="cb18-12"><a href="#cb18-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_viridis_c</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="Sepsis_rna_array_analysis_GSE65682_files/figure-html/unnamed-chunk-12-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
</section>
<section id="boxplots-of-specific-interactions-and-stats" class="level2">
<h2 class="anchored" data-anchor-id="boxplots-of-specific-interactions-and-stats">boxplots of specific interactions and stats</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb19"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb19-1"><a href="#cb19-1" aria-hidden="true" tabindex="-1"></a>interactions_plot <span class="ot">&lt;-</span> <span class="fu">pivot_longer</span>(rf_data, <span class="at">names_to =</span> <span class="st">"interaction"</span>, <span class="at">values_to =</span> <span class="st">"prob"</span>, <span class="dv">1</span><span class="sc">:</span><span class="dv">11518</span>)</span>
<span id="cb19-2"><a href="#cb19-2" aria-hidden="true" tabindex="-1"></a>top_features <span class="ot">&lt;-</span> vi_df[<span class="fu">order</span>(vi_df<span class="sc">$</span>Importance, <span class="at">decreasing =</span> <span class="cn">TRUE</span>)[<span class="dv">1</span><span class="sc">:</span><span class="dv">20</span>],]</span>
<span id="cb19-3"><a href="#cb19-3" aria-hidden="true" tabindex="-1"></a>interactions_plot <span class="ot">&lt;-</span> <span class="fu">filter</span>(interactions_plot, interaction <span class="sc">%in%</span> top_features<span class="sc">$</span>Gene)</span>
<span id="cb19-4"><a href="#cb19-4" aria-hidden="true" tabindex="-1"></a><span class="fu">options</span>(<span class="at">scipen =</span> <span class="dv">999</span>)</span>
<span id="cb19-5"><a href="#cb19-5" aria-hidden="true" tabindex="-1"></a>interactions_plot<span class="sc">$</span>mortality_28d <span class="ot">&lt;-</span> <span class="fu">factor</span>(interactions_plot<span class="sc">$</span>mortality_28d, <span class="at">levels =</span> <span class="fu">c</span>(<span class="st">"1"</span>, <span class="st">"0"</span>),</span>
<span id="cb19-6"><a href="#cb19-6" aria-hidden="true" tabindex="-1"></a>                                          <span class="at">labels =</span> <span class="fu">c</span>(<span class="st">"Non-survival"</span>, <span class="st">"Survival"</span>))</span>
<span id="cb19-7"><a href="#cb19-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-8"><a href="#cb19-8" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(interactions_plot, <span class="fu">aes</span>(<span class="at">x =</span> mortality_28d, <span class="at">y =</span> prob, <span class="at">fill =</span> mortality_28d))<span class="sc">+</span></span>
<span id="cb19-9"><a href="#cb19-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_boxplot</span>(<span class="at">outlier.shape =</span> <span class="cn">NA</span>, <span class="at">alpha =</span> <span class="fl">0.4</span>, <span class="at">colour =</span> <span class="st">"black"</span>)<span class="sc">+</span></span>
<span id="cb19-10"><a href="#cb19-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_jitter</span>(<span class="at">width =</span> <span class="fl">0.1</span>, <span class="at">shape =</span> <span class="dv">21</span>, <span class="at">colour =</span> <span class="st">"black"</span>, <span class="at">size =</span> <span class="dv">2</span>)<span class="sc">+</span></span>
<span id="cb19-11"><a href="#cb19-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_bw</span>()<span class="sc">+</span></span>
<span id="cb19-12"><a href="#cb19-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">facet_wrap</span>(<span class="sc">~</span>interaction, <span class="at">scales =</span> <span class="st">"free_y"</span>, <span class="at">ncol =</span> <span class="dv">4</span>)<span class="sc">+</span></span>
<span id="cb19-13"><a href="#cb19-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_manual</span>(<span class="at">values =</span> <span class="fu">c</span>(<span class="st">"darkorchid"</span>, <span class="st">"mediumseagreen"</span>))<span class="sc">+</span></span>
<span id="cb19-14"><a href="#cb19-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">xlab</span>(<span class="cn">NULL</span>)<span class="sc">+</span></span>
<span id="cb19-15"><a href="#cb19-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">axis.text =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">12</span>, <span class="at">colour =</span> <span class="st">"black"</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="Sepsis_rna_array_analysis_GSE65682_files/figure-html/unnamed-chunk-13-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="sourceCode cell-code" id="cb20"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb20-1"><a href="#cb20-1" aria-hidden="true" tabindex="-1"></a>results <span class="ot">&lt;-</span> interactions_plot <span class="sc">%&gt;%</span></span>
<span id="cb20-2"><a href="#cb20-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(interaction) <span class="sc">%&gt;%</span></span>
<span id="cb20-3"><a href="#cb20-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarise</span>(</span>
<span id="cb20-4"><a href="#cb20-4" aria-hidden="true" tabindex="-1"></a>    <span class="at">p_value =</span> <span class="fu">wilcox.test</span>(prob <span class="sc">~</span> mortality_28d)<span class="sc">$</span>p.value,</span>
<span id="cb20-5"><a href="#cb20-5" aria-hidden="true" tabindex="-1"></a>    <span class="at">.groups =</span> <span class="st">"drop"</span></span>
<span id="cb20-6"><a href="#cb20-6" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">%&gt;%</span></span>
<span id="cb20-7"><a href="#cb20-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(p_value)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
</section>

</main>
<!-- /main column -->
<script id="quarto-html-after-body" type="application/javascript">
  window.document.addEventListener("DOMContentLoaded", function (event) {
    const icon = "";
    const anchorJS = new window.AnchorJS();
    anchorJS.options = {
      placement: 'right',
      icon: icon
    };
    anchorJS.add('.anchored');
    const isCodeAnnotation = (el) => {
      for (const clz of el.classList) {
        if (clz.startsWith('code-annotation-')) {                     
          return true;
        }
      }
      return false;
    }
    const onCopySuccess = function(e) {
      // button target
      const button = e.trigger;
      // don't keep focus
      button.blur();
      // flash "checked"
      button.classList.add('code-copy-button-checked');
      var currentTitle = button.getAttribute("title");
      button.setAttribute("title", "Copied!");
      let tooltip;
      if (window.bootstrap) {
        button.setAttribute("data-bs-toggle", "tooltip");
        button.setAttribute("data-bs-placement", "left");
        button.setAttribute("data-bs-title", "Copied!");
        tooltip = new bootstrap.Tooltip(button, 
          { trigger: "manual", 
            customClass: "code-copy-button-tooltip",
            offset: [0, -8]});
        tooltip.show();    
      }
      setTimeout(function() {
        if (tooltip) {
          tooltip.hide();
          button.removeAttribute("data-bs-title");
          button.removeAttribute("data-bs-toggle");
          button.removeAttribute("data-bs-placement");
        }
        button.setAttribute("title", currentTitle);
        button.classList.remove('code-copy-button-checked');
      }, 1000);
      // clear code selection
      e.clearSelection();
    }
    const getTextToCopy = function(trigger) {
        const codeEl = trigger.previousElementSibling.cloneNode(true);
        for (const childEl of codeEl.children) {
          if (isCodeAnnotation(childEl)) {
            childEl.remove();
          }
        }
        return codeEl.innerText;
    }
    const clipboard = new window.ClipboardJS('.code-copy-button:not([data-in-quarto-modal])', {
      text: getTextToCopy
    });
    clipboard.on('success', onCopySuccess);
    if (window.document.getElementById('quarto-embedded-source-code-modal')) {
      const clipboardModal = new window.ClipboardJS('.code-copy-button[data-in-quarto-modal]', {
        text: getTextToCopy,
        container: window.document.getElementById('quarto-embedded-source-code-modal')
      });
      clipboardModal.on('success', onCopySuccess);
    }
      var localhostRegex = new RegExp(/^(?:http|https):\/\/localhost\:?[0-9]*\//);
      var mailtoRegex = new RegExp(/^mailto:/);
        var filterRegex = new RegExp('/' + window.location.host + '/');
      var isInternal = (href) => {
          return filterRegex.test(href) || localhostRegex.test(href) || mailtoRegex.test(href);
      }
      // Inspect non-navigation links and adorn them if external
     var links = window.document.querySelectorAll('a[href]:not(.nav-link):not(.navbar-brand):not(.toc-action):not(.sidebar-link):not(.sidebar-item-toggle):not(.pagination-link):not(.no-external):not([aria-hidden]):not(.dropdown-item):not(.quarto-navigation-tool):not(.about-link)');
      for (var i=0; i<links.length; i++) {
        const link = links[i];
        if (!isInternal(link.href)) {
          // undo the damage that might have been done by quarto-nav.js in the case of
          // links that we want to consider external
          if (link.dataset.originalHref !== undefined) {
            link.href = link.dataset.originalHref;
          }
        }
      }
    function tippyHover(el, contentFn, onTriggerFn, onUntriggerFn) {
      const config = {
        allowHTML: true,
        maxWidth: 500,
        delay: 100,
        arrow: false,
        appendTo: function(el) {
            return el.parentElement;
        },
        interactive: true,
        interactiveBorder: 10,
        theme: 'quarto',
        placement: 'bottom-start',
      };
      if (contentFn) {
        config.content = contentFn;
      }
      if (onTriggerFn) {
        config.onTrigger = onTriggerFn;
      }
      if (onUntriggerFn) {
        config.onUntrigger = onUntriggerFn;
      }
      window.tippy(el, config); 
    }
    const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
    for (var i=0; i<noterefs.length; i++) {
      const ref = noterefs[i];
      tippyHover(ref, function() {
        // use id or data attribute instead here
        let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
        try { href = new URL(href).hash; } catch {}
        const id = href.replace(/^#\/?/, "");
        const note = window.document.getElementById(id);
        if (note) {
          return note.innerHTML;
        } else {
          return "";
        }
      });
    }
    const xrefs = window.document.querySelectorAll('a.quarto-xref');
    const processXRef = (id, note) => {
      // Strip column container classes
      const stripColumnClz = (el) => {
        el.classList.remove("page-full", "page-columns");
        if (el.children) {
          for (const child of el.children) {
            stripColumnClz(child);
          }
        }
      }
      stripColumnClz(note)
      if (id === null || id.startsWith('sec-')) {
        // Special case sections, only their first couple elements
        const container = document.createElement("div");
        if (note.children && note.children.length > 2) {
          container.appendChild(note.children[0].cloneNode(true));
          for (let i = 1; i < note.children.length; i++) {
            const child = note.children[i];
            if (child.tagName === "P" && child.innerText === "") {
              continue;
            } else {
              container.appendChild(child.cloneNode(true));
              break;
            }
          }
          if (window.Quarto?.typesetMath) {
            window.Quarto.typesetMath(container);
          }
          return container.innerHTML
        } else {
          if (window.Quarto?.typesetMath) {
            window.Quarto.typesetMath(note);
          }
          return note.innerHTML;
        }
      } else {
        // Remove any anchor links if they are present
        const anchorLink = note.querySelector('a.anchorjs-link');
        if (anchorLink) {
          anchorLink.remove();
        }
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(note);
        }
        if (note.classList.contains("callout")) {
          return note.outerHTML;
        } else {
          return note.innerHTML;
        }
      }
    }
    for (var i=0; i<xrefs.length; i++) {
      const xref = xrefs[i];
      tippyHover(xref, undefined, function(instance) {
        instance.disable();
        let url = xref.getAttribute('href');
        let hash = undefined; 
        if (url.startsWith('#')) {
          hash = url;
        } else {
          try { hash = new URL(url).hash; } catch {}
        }
        if (hash) {
          const id = hash.replace(/^#\/?/, "");
          const note = window.document.getElementById(id);
          if (note !== null) {
            try {
              const html = processXRef(id, note.cloneNode(true));
              instance.setContent(html);
            } finally {
              instance.enable();
              instance.show();
            }
          } else {
            // See if we can fetch this
            fetch(url.split('#')[0])
            .then(res => res.text())
            .then(html => {
              const parser = new DOMParser();
              const htmlDoc = parser.parseFromString(html, "text/html");
              const note = htmlDoc.getElementById(id);
              if (note !== null) {
                const html = processXRef(id, note);
                instance.setContent(html);
              } 
            }).finally(() => {
              instance.enable();
              instance.show();
            });
          }
        } else {
          // See if we can fetch a full url (with no hash to target)
          // This is a special case and we should probably do some content thinning / targeting
          fetch(url)
          .then(res => res.text())
          .then(html => {
            const parser = new DOMParser();
            const htmlDoc = parser.parseFromString(html, "text/html");
            const note = htmlDoc.querySelector('main.content');
            if (note !== null) {
              // This should only happen for chapter cross references
              // (since there is no id in the URL)
              // remove the first header
              if (note.children.length > 0 && note.children[0].tagName === "HEADER") {
                note.children[0].remove();
              }
              const html = processXRef(null, note);
              instance.setContent(html);
            } 
          }).finally(() => {
            instance.enable();
            instance.show();
          });
        }
      }, function(instance) {
      });
    }
        let selectedAnnoteEl;
        const selectorForAnnotation = ( cell, annotation) => {
          let cellAttr = 'data-code-cell="' + cell + '"';
          let lineAttr = 'data-code-annotation="' +  annotation + '"';
          const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
          return selector;
        }
        const selectCodeLines = (annoteEl) => {
          const doc = window.document;
          const targetCell = annoteEl.getAttribute("data-target-cell");
          const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
          const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
          const lines = annoteSpan.getAttribute("data-code-lines").split(",");
          const lineIds = lines.map((line) => {
            return targetCell + "-" + line;
          })
          let top = null;
          let height = null;
          let parent = null;
          if (lineIds.length > 0) {
              //compute the position of the single el (top and bottom and make a div)
              const el = window.document.getElementById(lineIds[0]);
              top = el.offsetTop;
              height = el.offsetHeight;
              parent = el.parentElement.parentElement;
            if (lineIds.length > 1) {
              const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
              const bottom = lastEl.offsetTop + lastEl.offsetHeight;
              height = bottom - top;
            }
            if (top !== null && height !== null && parent !== null) {
              // cook up a div (if necessary) and position it 
              let div = window.document.getElementById("code-annotation-line-highlight");
              if (div === null) {
                div = window.document.createElement("div");
                div.setAttribute("id", "code-annotation-line-highlight");
                div.style.position = 'absolute';
                parent.appendChild(div);
              }
              div.style.top = top - 2 + "px";
              div.style.height = height + 4 + "px";
              div.style.left = 0;
              let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
              if (gutterDiv === null) {
                gutterDiv = window.document.createElement("div");
                gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
                gutterDiv.style.position = 'absolute';
                const codeCell = window.document.getElementById(targetCell);
                const gutter = codeCell.querySelector('.code-annotation-gutter');
                gutter.appendChild(gutterDiv);
              }
              gutterDiv.style.top = top - 2 + "px";
              gutterDiv.style.height = height + 4 + "px";
            }
            selectedAnnoteEl = annoteEl;
          }
        };
        const unselectCodeLines = () => {
          const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
          elementsIds.forEach((elId) => {
            const div = window.document.getElementById(elId);
            if (div) {
              div.remove();
            }
          });
          selectedAnnoteEl = undefined;
        };
          // Handle positioning of the toggle
      window.addEventListener(
        "resize",
        throttle(() => {
          elRect = undefined;
          if (selectedAnnoteEl) {
            selectCodeLines(selectedAnnoteEl);
          }
        }, 10)
      );
      function throttle(fn, ms) {
      let throttle = false;
      let timer;
        return (...args) => {
          if(!throttle) { // first call gets through
              fn.apply(this, args);
              throttle = true;
          } else { // all the others get throttled
              if(timer) clearTimeout(timer); // cancel #2
              timer = setTimeout(() => {
                fn.apply(this, args);
                timer = throttle = false;
              }, ms);
          }
        };
      }
        // Attach click handler to the DT
        const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
        for (const annoteDlNode of annoteDls) {
          annoteDlNode.addEventListener('click', (event) => {
            const clickedEl = event.target;
            if (clickedEl !== selectedAnnoteEl) {
              unselectCodeLines();
              const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
              if (activeEl) {
                activeEl.classList.remove('code-annotation-active');
              }
              selectCodeLines(clickedEl);
              clickedEl.classList.add('code-annotation-active');
            } else {
              // Unselect the line
              unselectCodeLines();
              clickedEl.classList.remove('code-annotation-active');
            }
          });
        }
    const findCites = (el) => {
      const parentEl = el.parentElement;
      if (parentEl) {
        const cites = parentEl.dataset.cites;
        if (cites) {
          return {
            el,
            cites: cites.split(' ')
          };
        } else {
          return findCites(el.parentElement)
        }
      } else {
        return undefined;
      }
    };
    var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
    for (var i=0; i<bibliorefs.length; i++) {
      const ref = bibliorefs[i];
      const citeInfo = findCites(ref);
      if (citeInfo) {
        tippyHover(citeInfo.el, function() {
          var popup = window.document.createElement('div');
          citeInfo.cites.forEach(function(cite) {
            var citeDiv = window.document.createElement('div');
            citeDiv.classList.add('hanging-indent');
            citeDiv.classList.add('csl-entry');
            var biblioDiv = window.document.getElementById('ref-' + cite);
            if (biblioDiv) {
              citeDiv.innerHTML = biblioDiv.innerHTML;
            }
            popup.appendChild(citeDiv);
          });
          return popup.innerHTML;
        });
      }
    }
  });
  </script>
</div> <!-- /content -->




</body></html>